### PR TITLE
Add SCEP (beta) provisioner support

### PR DESCRIPTION
**Pull request by hslatman**
This PR adds support for adding and updating SCEP provisioners.

It also includes some small typo fixes.

### CODE REVIEW

File: command/ca/provisionerbeta/add.go [hunk 1]
```diff
@@ -73,7 +73,13 @@ func addCommand() cli.Command {
 **step beta ca provisioner add** <name> **--type**=ACME [**--force-cn**]
 [**--admin-cert**=<file>] [**--admin-key**=<file>] [**--admin-provisioner**=<name>]
 [**--admin-subject**=<subject>] [**--password-file**=<file>] [**--ca-url**=<uri>]
-[**--root**=<file>] [**--context**=<name>]`,
+[**--root**=<file>] [**--context**=<name>]
+
+**step beta ca provisioner add** <name> **--type**=SCEP [**--force-cn**] [**--challenge**=<challenge>] 
+[**--capabilities**=<capabilities>] [**--include-root**] [**--minimum-public-key-length**=<length>]
```
Comment (dopey) [inline command/ca/provisionerbeta/add.go:79]: It think we can use `min-public-key-length`.

File: command/ca/provisionerbeta/provisioner.go [hunk 2]
```diff
@@ -170,6 +170,35 @@ var (
 		Usage: `Always set the common name in provisioned certificates.`,
 	}
 
+	// SCEP provisioner flags
```
Comment (dopey) [inline command/ca/provisionerbeta/provisioner.go:173]: When defining flags that have a value try to us the type in the description inside `<>`, like so: https://github.com/smallstep/cli/blob/6b23538520fc8c720b067bec2fb7fe630f8825f4/flags/flags.go#L251-L257. That's become sort of a standard.

Comment (dopey) [reply command/ca/provisionerbeta/provisioner.go:173]: According to the usage at the top, the type for the challenge flag is `<challenge>`.

File: command/ca/provisionerbeta/update.go [hunk 3]
```diff
@@ -794,3 +815,32 @@ func updateGCPDetails(ctx *cli.Context, p *linkedca.Provisioner) error {
...
+	}
+	details := data.SCEP
+
+	if ctx.IsSet("force-cn") {
+		details.ForceCn = ctx.Bool("force-cn")
+	}
+	if ctx.IsSet("challenge") {
+		details.Challenge = ctx.String("challenge")
+	}
+	if ctx.IsSet("capabilities") {
```
Comment (dopey) [inline command/ca/provisionerbeta/update.go:832]: Are you intending to replace the entire list with the new one? For a few of the other provisioners I had to define a `remove-<flag>` to have a list of the values to remove and then do the logic in the update method before post-ing. Might no be necessary for this, just letting you know.

File: command/ca/provisionerbeta/update.go [hunk 4]
```diff
@@ -794,3 +815,32 @@ func updateGCPDetails(ctx *cli.Context, p *linkedca.Provisioner) error {
...
+		details.Capabilities = ctx.StringSlice("capabilities")
+	}
+	if ctx.IsSet("minimum-public-key-length") {
```
Comment (dopey) [inline command/ca/provisionerbeta/update.go:835]: don't forget to change this to `min` if you change the flag name

**Review by dopey — CHANGES_REQUESTED** [review]
see comments

File: command/ca/provisionerbeta/add.go (hunk 1 above)
Comment (hslatman) [reply command/ca/provisionerbeta/add.go]: Indeed; more consistent with other minimums. Fixed in 8606fdcf0560e56688309948f5c81d23f8588022.

File: command/ca/provisionerbeta/provisioner.go (hunk 2 above)
Comment (hslatman) [reply command/ca/provisionerbeta/provisioner.go:173]: 👍 

Will try to keep this in mind for future additions.

File: command/ca/provisionerbeta/update.go (hunk 3 above)
Comment (hslatman) [reply command/ca/provisionerbeta/update.go:832]: I saw your previous work and thought about adding those, but decided against it (for now). There'll likely be more work in the future related to SCEP and that might also touch the capabilities. Making this part a bit more user friendly can be done then, I think.

File: command/ca/provisionerbeta/update.go (hunk 4 above)
Comment (hslatman) [reply command/ca/provisionerbeta/update.go]: Fixed in 8606fdcf0560e56688309948f5c81d23f8588022.

**Review by dopey — APPROVED** [review]
lgtm

**Pull request merged**